### PR TITLE
DO-239: Fixed validation issue on family details.

### PR DESCRIPTION
**Pull request by dhruvlahoti**

### CODE REVIEW

File: src/components/member/MemberDetailsWizard/CustomForm/FamilyDetailsForm/FamilyDetailsForm.jsx [hunk 1]
```diff
@@ -12,30 +12,38 @@ class FamilyDetailsForm extends React.Component {
...
+  componentDidMount() {
+    let updatedFormData = this.state.formData;
+    if ((this.props.formData).length) {
+      (this.props.formData).forEach((data, index) => {
+        Object.keys(data).forEach((key) => {
+          if (key === 'age' && data[key] === null) {
+            updatedFormData[index][key] = undefined;
+          } else if (data[key] === null) {
+            updatedFormData[index][key] = '';
+          }
```
Comment (ayushmalik2110) [inline src/components/member/MemberDetailsWizard/CustomForm/FamilyDetailsForm/FamilyDetailsForm.jsx:38]: We have a util function for this 'formatFormData' in member/utils that do the same job. Replace all null value with the undefined.

Comment (ayushmalik2110) [reply src/components/member/MemberDetailsWizard/CustomForm/FamilyDetailsForm/FamilyDetailsForm.jsx:38]: Check other form for reference.

Comment (dhruvlahoti) [reply src/components/member/MemberDetailsWizard/CustomForm/FamilyDetailsForm/FamilyDetailsForm.jsx]: Done

File: src/components/member/MemberDetailsWizard/CustomForm/FamilyDetailsForm/familyDetailsSchema.json [hunk 2]
```diff
@@ -81,5 +81,14 @@
...
     }
+  },
+  "fieldTypeMap":  {
+    "age": "number",
+    "dob": "string",
+    "name": "string",
+    "relationshipType": "string",
+    "gender": "string",
+    "education": "string",
+    "profession": "string"
```
Comment (ayushmalik2110) [inline src/components/member/MemberDetailsWizard/CustomForm/FamilyDetailsForm/familyDetailsSchema.json:92]: You don't have to define field type map. It is already created. And you already have that in your props. Just check it.
image

Comment (dhruvlahoti) [reply src/components/member/MemberDetailsWizard/CustomForm/FamilyDetailsForm/familyDetailsSchema.json:92]: Its not working if I don't pass fieldTypeMap

Comment (ayushmalik2110) [reply src/components/member/MemberDetailsWizard/CustomForm/FamilyDetailsForm/familyDetailsSchema.json:92]: I know this method won't work if you don't pass the field type map. What I am saying is that you don't have to create another field type map when it is already created. Check the detailsWizard file.

Comment (dhruvlahoti) [reply src/components/member/MemberDetailsWizard/CustomForm/FamilyDetailsForm/familyDetailsSchema.json]: Done

**Pull request merged**